libs/core/src/lib/dialog/dialog.component.ts

```diff
@@ -110,8 +115,17 @@ export class DialogComponent implements OnInit, AfterViewInit, OnDestroy, CssCla
         @Optional() @Inject(DIALOG_CONFIG) public dialogConfig: DialogConfig,
         @Optional() @Inject(DIALOG_REF) private _dialogRef: DialogRef,
         private _elementRef: ElementRef,
-        private _changeDetectorRef: ChangeDetectorRef
-    ) {}
+        private _changeDetectorRef: ChangeDetectorRef,
+        private _router: Router
+    ) {
+        this._subscriptions.add(
```

Comment:
- Do not implement component logic in constructor, use life-cycle hooks
- Move to separate method